projects/packages/compat/lib/locales.php

```diff
@@ -0,0 +1,2398 @@
+<?php // phpcs:disable
+// This file is copied over from GlotPress/GlotPress-WP project and does not nessitate linting.
+
+if ( ! class_exists( 'GP_Locale' ) ) :
+
+	class GP_Locale {
+		public $english_name;
+		public $native_name;
+		public $text_direction      = 'ltr';
+		public $lang_code_iso_639_1 = null;
+		public $lang_code_iso_639_2 = null;
+		public $lang_code_iso_639_3 = null;
+		public $country_code;
+		public $wp_locale;
+		public $slug;
+		public $nplurals                         = 2;
+		public $plural_expression                = 'n != 1';
+		public $google_code                      = null;
+		public $preferred_sans_serif_font_family = null;
+		public $facebook_locale                  = null;
+		// TODO: days, months, decimals, quotes
+
+		private $_index_for_number;
+
+		public function __construct( $args = array() ) {
+			foreach ( $args as $key => $value ) {
+				$this->$key = $value;
+			}
+		}
+
+		public static function __set_state( $state ) {
+			return new GP_Locale( $state );
+		}
+
+		/**
+		 * Make deprecated properties checkable for backwards compatibility.
+		 *
+		 * @param string $name Property to check if set.
+		 * @return bool Whether the property is set.
+		 */
+		public function __isset( $name ) {
+			if ( 'rtl' == $name ) {
+				return isset( $this->text_direction );
+			}
+		}
+
+		/**
+		 * Make deprecated properties readable for backwards compatibility.
+		 *
+		 * @param string $name Property to get.
+		 * @return mixed Property.
+		 */
+		public function __get( $name ) {
+			if ( 'rtl' == $name ) {
+				return ( 'rtl' === $this->text_direction );
+			}
+		}
+
+		public function combined_name() {
+			/* translators: combined name for locales: 1: name in English, 2: native name */
+			return sprintf( _x( '%1$s/%2$s', 'locales', 'jetpack' ), $this->english_name, $this->native_name );
```

Comment:
Should we update the textdomain here?

```suggestion
			return sprintf( _x( '%1$s/%2$s', 'locales', 'jetpack-compat' ), $this->english_name, $this->native_name );
```

Reply:
Good catch 👍